descqa/ImgPkTest.py

```diff
@@ -44,49 +40,47 @@ def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
             return TestResult(skipped=True, summary='invalid rebinning value: {}'.format(rebinning))
         if len(test_raft.sensors) != 9:
             return TestResult(skipped=True, summary='Raft is not complete')
-        xdim, ydim = first(test_raft.sensors.values()).get_data().shape
-        
+
         # Assemble the 3 x 3 raft's image
         # TODO: Need to use LSST's software to handle the gaps properly
         total_data = np.array([test_raft.sensors['S%d%d'%(i, j)].get_data() for i in range (3) for j in range(3)])
-        total_data = total_data.reshape(3,3,xdim,ydim).swapaxes(1,2).reshape(3*xdim,3*ydim)
+        xdim, ydim = total_data.shape[1:]
+        total_data = total_data.reshape(3, 3, xdim, ydim).swapaxes(1, 2).reshape(3*xdim, 3*ydim)
+
         # FFT of the density contrast
-        F1 = fftpack.fft2((total_data/np.mean(total_data)-1))
-        F2 = fftpack.fftshift( F1 )
-        psd2D = np.abs( F2 )**2 # 2D power
-        pix_scale = 0.2/60*rebinning #pixel scale in arcmin 
-        kx = fftpack.fftshift(fftpack.fftfreq(F2.shape[0], pix_scale))
-        ky = fftpack.fftshift(fftpack.fftfreq(F2.shape[1], pix_scale))
-        kxx, kyy = np.meshgrid(kx, ky)
-        rad = np.sqrt(kxx**2 + kyy**2)
-        bins = fftpack.rfftfreq(F2.shape[0], pix_scale)
-        bin_space = bins[1]-bins[0]
-        ps1d = np.zeros(len(bins))
-        for i, b in enumerate(bins):
-            ps1d[i] = np.mean(
-                psd2D.T[(rad > b - 0.5 * bin_space) & (rad < b + 0.5 * bin_space)]) / (F2.shape[0] * F2.shape[1])
-        bins = bins / (2 * np.pi)
-        fig, ax = plt.subplots(2, 1)
+        FT = np.fft.fft2(total_data / total_data.mean() - 1)
+        n_kx, n_ky = FT.shape
+        psd2D = np.square(np.abs(FT)).ravel() # 2D power, flattened
+        psd2D /= (n_kx * n_ky)
+        pix_scale = 0.2 / 60.0 #pixel scale in arcmin
+        spacing = pix_scale * rebinning
+        rad = np.hypot(*np.meshgrid(np.fft.fftfreq(n_kx, spacing), np.fft.fftfreq(n_ky, spacing), indexing='ij')).ravel()
+        rad /= (2.0 * np.pi)
+
+        psd1D, bin_edges, _ = binned_statistic(rad, psd2D, bins=self.validation_data['k'])
+        bin_center = (bin_edges[1:] + bin_edges[:-1]) * 0.5
+        score = np.square(psd1D/self.validation_data['Pk'][:-1] - 1.0).sum()
```

Comment:
This assumes that the validation data has the same length and binning than the power-spectrum computed. I think it would be better to add either a `try, except` statement or interpolate the measured power-spectra.